tap_salesforce/salesforce/__init__.py

```diff
@@ -235,6 +240,7 @@ def __init__(self,
         self.login_timer = None
         self.data_url = "{}/services/data/v52.0/{}"
         self.pk_chunking = False
+        self.lookback_window = lookback_window
 
         # validate start_date
         singer_utils.strptime(default_start_date)
```

Comment:
Update this to `singer_utils.strptime_to_utc` to get unit tests working again
- `singer_utils.strptime` is deprecated

The unit tests didn't catch this because we mocked away this function. My test removed the Mock and allowed this to fail properly.

---

The difference between [singer_utils.strptime](https://github.com/singer-io/singer-python/blob/f2b383324a3a97b98d6e29f69dbadae2632dbd2f/singer/utils.py#L58) and [singer_utils.strptime_to_utc](https://github.com/singer-io/singer-python/blob/f2b383324a3a97b98d6e29f69dbadae2632dbd2f/singer/utils.py#L61) is the parsing function used.

`strptime` needs a datetime format to function correctly - which results in the error that you see in Circle

This is also backwards compatible assuming we only write bookmarks with `singer-python` functions. This ensures that bookmarks are all converted to UTC first. And `singer_utils.strptime_to_utc` would just work because it reads a variety of date time formats (with and without fractional seconds) and then converts it to UTC.

---

Example to show that we can parse fractional seconds:
```python
>>> import dateutil.parser
>>> dateutil.parser.parse("2022-05-02T00:00:00.000000Z")
datetime.datetime(2022, 5, 2, 0, 0, tzinfo=tzlocal())
>>> dateutil.parser.parse("2022-05-02T00:00:00Z")
datetime.datetime(2022, 5, 2, 0, 0, tzinfo=tzlocal())
```

Reply:
Thanks for the information. Updated `strptime` to `strptime_to_utc` and we will use **strptime_to_utc** from now onwards when needed.